base/libgit2/repository.jl

```diff
@@ -181,23 +174,25 @@ function path(repo::GitRepo)
     end
 end
 
-function peel(obj::GitObject, obj_type::Cint)
-    peeled_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
-    git_otype = getobjecttype(obj_type)
-    err = ccall((:git_object_peel, :libgit2), Cint,
-                (Ptr{Ptr{Void}}, Ptr{Void}, Cint), peeled_ptr_ptr, obj.ptr, obj_type)
-    if err == Int(Error.ENOTFOUND)
-        return GitHash()
-    elseif err != Int(Error.GIT_OK)
-        if peeled_ptr_ptr[] != C_NULL
-            close(GitUnknownObject(obj.repo, peeled_ptr_ptr[]))
-        end
-        throw(Error.GitError(err))
-    end
-    return git_otype(obj.repo, peeled_ptr_ptr[])
+"""
+    peel([T,] obj::GitObject)
+
+Recursively peel `obj` until an object of type `T` is obtained. If no `T` is provided,
+then it will be peeled until the type changes.
```

Comment:
can we put `obj` instead of "it" here, otherwise it's potentially ambiguous whether "it" refers to `T` or `obj`.